test/library/packages/Shlex/Shlex.chpl

```diff
@@ -0,0 +1,385 @@
+        var tok=this.pushback.pop(0);
+        if this.debug >=1 {
+          writeln("shlex: popping token %s\n", tok);
+        }
+        return tok;
+      }
+      if(this.tokindex==-1){
+        return '';
+      }
+      var raw= try! this.read_token();
+      if this.debug>=1{
+        if(raw!=''){
+          writeln("shlex: token="+raw);
+        }
+        else{
+          writeln("shlex: token=EOF");
+        }
+      }
+      return raw;
+    }
+
+    //
+    // Reads next token and returns it
+    //
+    proc read_token() throws{
+      var quoted:bool = false;
+      var escapsedstate:string = ' ';
+      var nextchar:string='';
+      while true {
+        if this.tokindex==-1 then break;
+        if (this.punctuation_chars!='' && !this._pushback_chars.isEmpty()){
+          nextchar=this._pushback_chars.pop();
+        }
+        else if(this.tokindex<this.instream.size){
+          nextchar=this.instream[this.tokindex];
+          this.tokindex+=1;
+        }
+        else{
+          nextchar='';
+          this.tokindex=-1;
+        }
+        if this.debug>=3 then writeln("shlex: in state ", this.state,
+                                        "I see character: ", nextchar);
+        if nextchar=='\n' then this.lineno+=1;
+        if this.state==''{
+          this.token = '';
+          break;
+        }
+        else if this.state==' '{
+          if nextchar=='' {
+            this.state='';
+            break;
+          }
+          else if(this.whitespace.find(nextchar)!=-1){
+            if(this.debug>=2){
+              writeln("shlex: I see whitespaces in whitespace state");
+            }
+            if (this.token!='' || (this.posix && quoted)) then break;
+            else continue;
+          }
+          else if(this.commenters.find(nextchar)!=-1){
+            while(this.tokindex<this.instream.size &&
+                    this.instream[tokindex]!='\n') do
+              this.tokindex+=1;
+            this.lineno+=1;
+          }
+          else if(this.posix && this.escapse.find(nextchar)!=-1){
+            escapsedstate='a';
+            this.state=nextchar;
+          }
+          else if(this.wordchars.find(nextchar)!=-1){
+            this.token=nextchar;
+            this.state='a';
+          }
+          else if(this.punctuation_chars.find(nextchar)!=-1){
+            this.token=nextchar;
+            this.state='c';
+          }
+          else if(this.quotes.find(nextchar)!=-1){
+            if !this.posix then this.token=nextchar;
+            this.state=nextchar;
+          }
+          else if(this.whitespace_split){
+            this.token=nextchar;
+            this.state='a';
+          }
+          else{
+            this.token=nextchar;
+            if (this.token!='' || (this.posix && quoted)) then break;
+            else continue;
+          }
+        }
+        else if (this.quotes.find(this.state)!=-1){
+          quoted=true;
+          if nextchar==''{
+            if this.debug>=2{
```

Comment:
formatting nit: I think the spaces here are generally more readable (and I think this is what @rapiz1 was suggesting):

```suggestion
            if this.debug >= 2 {
```